src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/ingredient/UniqueIngredientList.java

```diff
@@ -0,0 +1,98 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.ingredient;
+
+import static ay2122s1_cs2103t_w16_2.btbb.commons.util.CollectionUtil.requireAllNonNull;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+/**
+ * A list of ingrdients that enforces uniqueness between its elements and does not allow nulls.
+ * An ingredient is considered unique by comparing using {@code Ingredient#isSameIngredient(Ingredient)}.
+ * As such, adding and updating of ingredients uses Ingredient#isSameIngredient(Ingredient) for equality so as to
+ * ensure that the ingredient being added or updated is unique in terms of identity in the UniqueIngredientList.
+ * However, the removal of an ingredent uses Ingredient#equals(Object) so as to ensure that the ingredient with exactly
+ * the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Ingredient#isSameIngredient(Ingredient)
+ */
+public class UniqueIngredientList implements Iterable<Ingredient> {
+
+    private final ObservableList<Ingredient> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Ingredient> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
```

Comment:
Remove the newline at line 25. No gap between class declaration and first member.